mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb

```diff
@@ -1003,6 +1003,112 @@ func TestAccContainerCluster_withNodeConfig(t *testing.T) {
 	})
 }
 
+func TestAccContainerCluster_withNoSpecifiedLoggingVariant(t *testing.T) {
+	t.Parallel()
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_basic(clusterName),
+				Check: resource.TestCheckResourceAttr("google_container_cluster.primary",
+					"node_pool_defaults.0.node_config_defaults.0.logging_variant",
+					"DEFAULT"),
+			},
+			{
+				ResourceName:      "google_container_cluster.primary",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccContainerCluster_withDefaultLoggingVariant(t *testing.T) {
+	t.Parallel()
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_withLoggingVariantNodePoolDefault(clusterName, "DEFAULT"),
+				Check: resource.TestCheckResourceAttr("google_container_cluster.with_logging_variant_node_pool_default",
+					"node_pool_defaults.0.node_config_defaults.0.logging_variant",
+					"DEFAULT"),
+			},
+			{
+				ResourceName:      "google_container_cluster.with_logging_variant_node_pool_default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccContainerCluster_withMaxThroughputLoggingVariant(t *testing.T) {
+	t.Parallel()
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_withLoggingVariantNodePoolDefault(clusterName, "MAX_THROUGHPUT"),
+				Check: resource.TestCheckResourceAttr("google_container_cluster.with_logging_variant_node_pool_default",
+					"node_pool_defaults.0.node_config_defaults.0.logging_variant",
+					"MAX_THROUGHPUT"),
+			},
+			{
+				ResourceName:      "google_container_cluster.with_logging_variant_node_pool_default",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccContainerCluster_withLoggingVariantUpdates(t *testing.T) {
```

Comment:
Probably only need this test for container cluster node_pool_default - however, you'll also need to test the following (either in this test or in a separate one):

- google_container_cluster.node_pool.node_config.logging_variant
- google_container_cluster.node_config.logging_variant

Reply:
I created 2 tests for these 2 scenarios, kept the update test, and deleted the others.